worldSetup.py

```diff
@@ -25,31 +39,21 @@ def __init__(self):
         self.bingo_goals = None
         self.multichest = RUPEES_20
 
-    def randomize(self, settings, rnd):
-        if settings.overworld == "dungeondive":
-            self.entrance_mapping = {"d%d" % (n): "d%d" % (n) for n in range(9)}
-        if settings.randomstartlocation and settings.entranceshuffle == "none":
-            # List of all the possible locations where we can place our starting house
-            start_locations = [
-                "phone_d8",
-                "rooster_house",
-                "writes_phone",
-                "castle_phone",
-                "photo_house",
-                "start_house",
-                "prairie_right_phone",
-                "banana_seller",
-                "prairie_low_phone",
-                "animal_phone",
-            ]
-            start_location = start_locations.pop(rnd.randrange(len(start_locations)))
-            if start_location != "start_house":
-                self.entrance_mapping[start_location] = "start_house"
-                self.entrance_mapping["start_house"] = start_location
+    def getEntrancePool(self, settings, connectorsOnly=False):
+        entrances = []
+
+        if connectorsOnly:
+            if settings.entranceshuffle in ("advanced", "expert", "insanity"):
+                entrances = [k for k, v in ENTRANCE_INFO.items() if v.type == "connector"]
+            
+            return entrances
+
+        tradeEntrances = ['writes_house', 'banana_seller', 'animal_house5', 'animal_house3']
```

Comment:
Actually, instead of modifying the type. I would just add a new type called "trade" or something, and build up the entrance list based on which settings are enabled or not. You're now modifying a global, which will mess with multiworld.

Reply:
Good point, that makes more sense all around. Just pushed an update.